compiler/annotation/function.rs

```diff
@@ -48,10 +54,18 @@ pub struct AnnotatedFunction {
     pub variable_registry: VariableRegistry,
     pub parameter_registry: ParameterRegistry,
     pub arguments: Vec<Variable>,
+    pub argument_annotations: Vec<FunctionParameterAnnotation>,
     pub stages: Vec<AnnotatedStage>,
     pub return_: AnnotatedFunctionReturn,
 }
 
+impl AnnotatedFunction {
+    pub(crate) fn get_annotated_signature(&self) -> AnnotatedFunctionSignature {
```

Comment:
so i general, im not a huge fan of cloning things as they are returned - better to use Refs (Cows/maybeOwned) and clone on the receiver end when you need it to be owned

Reply:
Or store an AnnotatedFunctionSignature and pass an & back, then clone that if needed

Reply:
Reworked the whole thing. Please re-review the structure of `AnnotatedFunction`